### PR TITLE
Add ability to respond to webhooks.

### LINKED ISSUE #837: Hookshot does not correctly respond to Salesforce webhooks
Salesforce [requires](https://help.salesforce.com/s/articleView?id=000381752&type=1) that you respond this to any webhook, otherwise it assumes not received and resends. This causes Hookshot to post the same events repeatably every time Sf retries

```xml
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
<soapenv:Body>
<notificationsResponse xmlns="http://soap.sforce.com/2005/09/outbound">
<Ack>true</Ack>
</notificationsResponse>
</soapenv:Body>
</soapenv:Envelope>
```

The customer also sees this in their Salesforce webhook log
> org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog

Nothing useful in the Hookshot logs with debug enabled.

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #837 

This adds two new features to the generic webhook connection type:
 - The ability to set a response value for webhooks.
 - The ability for connections to choose whether to block on the webhook function before responding.

### CODE REVIEW

**Comment by Twi1ightSparkle** [comment]
Hookshot is posting the message twice to Matrix both with and without `Wait for webhook function to complete before responding` checked. But looks to be sending the correct data back.

```javascript
const responseBody = `
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
<soapenv:Body>
<notificationsResponse xmlns="http://soap.sforce.com/2005/09/outbound">
<Ack>true</Ack>
</notificationsResponse>
</soapenv:Body>
</soapenv:Envelope>
`.trim();

result = {
    empty: false,
    msgtype: 'm.text',
    plain: message,
    version: 'v2',
    webhookResponse: {
        body: responseBody,
        contentType: 'application/xml',
        statusCode: 418,
    }
};
```

Screenshot 2023-11-08 at 5 41 52 PM

---

Also, `Wait for webhook function to complete before responding` unchecks itself when you close and re-open the webhook in the configurator.

---

Waiting for my contact to do an actual test with Salesforce.

**Comment by Twi1ightSparkle** [comment]
Actually, once you have enabled `Wait for webhook function to complete before responding.`, you can't turn it off again. It appears to just not read `waitForComplete": true` from the `"uk.half-shot.matrix-hookshot.generic.hook",` state event when opening the webhook

**Review by AndrewFerr — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+4.5.1 (2023-11-09)
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply CHANGELOG.md:1]: Well spotted, how did that happen :/

File: docs/setup/webhooks.md [hunk 2]
```diff
@@ -101,6 +102,15 @@ to a string representation of that value. This change is <strong>not applied</st
 variable, so it will contain proper float values.
 </section>
 
+### Wait for complete
+
+It is possible to choose whether a webhook response should be instant, or after hookshot has handled the message. The reason
+for this is that some services expect a quick response time (Slack) whereas others will wait for the request to complete. You
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply docs/setup/webhooks.md:108]: I meant Slack, but I should probably make that clearer.

**Review by AndrewFerr — APPROVED** [review]

**Pull request merged**